umbra/umbra_32.c

```diff
@@ -1049,3 +1049,53 @@ umbra_handle_fault(void *drcontext, byte *target, dr_mcontext_t *raw_mc,
 {
     return false;
 }
+
+drmf_status_t
+umbra_clear_redundant_blocks(umbra_map_t *map, uint *count)
+{
+    byte *shadow_data;
+    uint i, j;
+
+    if (map == NULL)
+        return DRMF_ERROR_INVALID_PARAMETER;
+
+    if (count != NULL)
+        *count = 0;
+
+    /* Umbra needs to have create-on-touch optimization enabled. */
+    if (!TEST(UMBRA_MAP_CREATE_SHADOW_ON_TOUCH, map->options.flags)){
```

Comment:
I think the doxygen doc should state that it will fail w/o this option.

Reply:
Done.